src/olympia/addons/views.py

```diff
@@ -224,6 +230,13 @@ def check_object_permissions(self, request, obj):
                 'is_disabled_by_mozilla': obj.status == amo.STATUS_DISABLED,
             }
             raise exc
+        region_code = (
+            self.request and self.request.META.get(
+                'HTTP_X_COUNTRY_CODE', None))
+        if region_code and AddonRegionalRestrictions.objects.filter(
+                addon=obj,
+                excluded_regions__contains=region_code.upper()).exists():
```

Comment:
Once we start getting this for all requests it'll probably be more efficient to use `select_related` on `'regional_restrictions'` vs. a separate query (as a follow-up if so)

Reply:
Also, shouldn't we use `icontains` here?

Reply:
Doing `icontains` would make the query a ` LIKE %` (which could lead to false positives) instead of a `JSON_CONTAINS()`

Reply:
Interesting. how do we ensure that entries will be correct in the JSON then?

Reply:
a big stick gestured at whoever updates the data 😜

Reply:
Something to ensure in https://github.com/mozilla/addons-server/issues/15968 in the long run